worlds/dark_souls_3/__init__.py

```diff
@@ -0,0 +1,258 @@
+        set_rule(self.world.get_entrance("Goto Irithyll of the boreal", self.player),
+                 lambda state: state.has("Small Doll", self.player))
+        set_rule(self.world.get_entrance("Goto Archdragon peak", self.player),
+                 lambda state: state.has("Path of the Dragon Gesture", self.player))
+        set_rule(self.world.get_entrance("Goto Profaned capital", self.player),
+                 lambda state: state.has("Storm Ruler", self.player))
+        set_rule(self.world.get_entrance("Goto Grand Archives", self.player),
+                 lambda state: state.has("Grand Archives Key", self.player))
+        set_rule(self.world.get_entrance("Goto Kiln Of The First Flame", self.player),
+                 lambda state: state.has("Cinders of a Lord - Abyss Watcher", self.player) and
+                 state.has("Cinders of a Lord - Yhorm the Giant", self.player) and
+                 state.has("Cinders of a Lord - Aldrich", self.player) and
+                 state.has("Cinders of a Lord - Lothric Prince", self.player))
+
+        self.world.completion_condition[self.player] = lambda state: \
+            state.has("Cinders of a Lord - Abyss Watcher", self.player) and \
+            state.has("Cinders of a Lord - Yhorm the Giant", self.player) and \
+            state.has("Cinders of a Lord - Aldrich", self.player) and \
+            state.has("Cinders of a Lord - Lothric Prince", self.player)
+
+    def generate_basic(self):
+        # Depending on the specified option, add the Basin of Vows to a specific location or to the item pool
+        item = self.create_item("Basin of Vows")
+        if self.world.late_basin_of_vows[self.player]:
+            self.world.get_location("IBV: Soul of Pontiff Sulyvahn", self.player).place_locked_item(item)
+        else:
+            self.world.itempool += [item]
+
+    def generate_output(self, output_directory: str):
+        # Depending on the specified option, modify items hexadecimal value to add an upgrade level
+        item_dictionary = item_dictionary_table.copy()
+        if self.world.randomize_weapons_level[self.player]:
+            # Randomize some weapons upgrades
+            for name in weapons_upgrade_5_table.keys():
+                if self.world.random.randint(0, 100) < 33:
+                    value = self.world.random.randint(1, 5)
+                    item_dictionary[name] += value
+
+            for name in weapons_upgrade_10_table.keys():
+                if self.world.random.randint(0, 100) < 33:
+                    value = self.world.random.randint(1, 10)
+                    item_dictionary[name] += value
+
+        # Create the mandatory lists to generate the player's output file
+        items_id = []
+        items_address = []
+        locations_id = []
+        locations_address = []
+        locations_target = []
+        for location in self.world.get_filled_locations():
+            if location.item.player == self.player:
```

Comment:
This makes me think this will break with item_links

Reply:
I just set the item_links as follow for 2 players : 
```
item_links:
   - name: goMode
     item_pool: 
      - Cinders of a Lord - Abyss Watcher
      - Cinders of a Lord - Yhorm the Giant
      - Cinders of a Lord - Aldrich
      - Cinders of a Lord - Lothric Prince
     replacement_item: null 
```

And the nominal case seems to work as expected as the two players received the item : 
```
Server :         (Team #1) MarechToo sent Cinders of a Lord - Abyss Watcher to goMode (HWL: Longbow
Marech's client :   "#0: Cinders of a Lord - Abyss Watcher from MarechToo - HWL: Longbow\n",
MarechToo's client : "#0: Cinders of a Lord - Abyss Watcher from MarechToo - HWL: Longbow\n"
```

I would need help to list all different cases possible and an entire run to test them all.